orthoinference/src/main/java/org/reactome/orthoinference/OrthologousEntityGenerator.java

```diff
@@ -26,84 +26,86 @@
 	private static GKInstance complexSummationInst;
 	private static GKInstance speciesInst;
 	private static GKInstance nullInst = null;
-	private static Map<GKInstance, GKInstance> orthologousEntityIdenticals = new HashMap<GKInstance,GKInstance>();
-	private static Map<GKInstance, GKInstance> homolEWASIdenticals = new HashMap<GKInstance,GKInstance>();
-	private static Map<GKInstance, GKInstance> complexPolymerIdenticals = new HashMap<GKInstance, GKInstance>();
-	private static Map<GKInstance, GKInstance> inferredEntitySetIdenticals = new HashMap<GKInstance, GKInstance>();
-	private static Map<String,GKInstance> definedSetIdenticals = new HashMap<String,GKInstance>();
-	private static Map<String,GKInstance> complexIdenticals = new HashMap<String,GKInstance>();
-	private static Map<String,GKInstance> entitySetIdenticals = new HashMap<String,GKInstance>();
+	private static Map<GKInstance, GKInstance> orthologousEntityIdenticals = new HashMap<>();
+	private static Map<GKInstance, GKInstance> homolEWASIdenticals = new HashMap<>();
+	private static Map<GKInstance, GKInstance> complexPolymerIdenticals = new HashMap<>();
+	private static Map<GKInstance, GKInstance> inferredEntitySetIdenticals = new HashMap<>();
+	private static Map<String,GKInstance> definedSetIdenticals = new HashMap<>();
+	private static Map<String,GKInstance> complexIdenticals = new HashMap<>();
+	private static Map<String,GKInstance> entitySetIdenticals = new HashMap<>();
 
 /** The heart of the OrthoInference process. This function takes PhysicalEntity (PE) instances and will infer those that are EWAS', Complexes/Polymers, or EntitySets.
 	 The function's arguments are an incoming PE instance and an override attribute. Instances that are comprised of PE's will often recursively call this createOrthoEntity function
 	 on constituent PE's with the override attribute set to 'true'. This ensures that these PE's are inferred, despite the fact that they might not pass some filter criteria.
 	 This is often handled using 'mock' instances (i.e. 'ghost instances' from Perl script), which allow a PE to be inferred without having to commit a 'real' instance to the DB.
 */
-	public static GKInstance createOrthoEntity(GKInstance entityInst, boolean override) throws InvalidAttributeException, Exception
+	public static GKInstance createOrthoEntity(GKInstance entityInst, boolean override) throws Exception
 	{
-		logger.info("\tGenerating orthologous instance for " + entityInst);
+		logger.info("Attempting PE inference: " + entityInst);
```

Comment:
This method has pretty deep nesting.  It would be easier to read if some of the outer conditions were made into guard clauses.

```
if (!entityInst.getSchemClass().isValidAttribute(species)) {
	// This used to have a conditional statement based on the returned value of the 'check_intracellular' function.
	// That function doesn't exist anymore (only seemed to apply to the 'mtub' species, which hasn't been inferred for a while).
	// Since the instance is species-agnostic, just returns the original instance.
	logger.info("Could not find valid species attribute, returning original instance: " + entityInst);
	return entityInst;
}

if (orthologousEntityIdenticals.get(entityInst) != null) {
	logger.info("Inferred PE instance already exists");
	return orthologousEntityIdenticals.get(entityInst);
}
```